src/common/string.cpp

```diff
@@ -1677,6 +1677,24 @@ int wxString::Find(wxUniChar ch, bool bFromEnd) const
     *pVal = val;                                                            \
     return !*end;
 
+bool wxString::ToInt(int *pVal, int base) const
+{
+    wxASSERT_MSG(!base || (base > 1 && base <= 36), wxT("invalid base"));
+
+    WX_STRING_TO_X_TYPE_START
+    int val = wxStrtol(start, &end, base);
```

Comment:
This is not right, as it silently truncates values non representable as int on LP64 platforms. There must be a check that the value is in `-INT_MAX..INT_MAX` range here.